src/main/kotlin/com/dunctebot/dashboard/controllers/SettingsController.kt

```diff
@@ -30,17 +30,36 @@ import com.dunctebot.dashboard.utils.fetchGuildPatronStatus
 import com.dunctebot.models.settings.GuildSetting
 import com.dunctebot.models.settings.WarnAction
 import com.dunctebot.models.utils.Utils.colorToInt
-import net.dv8tion.jda.api.entities.Guild
 import spark.Request
 import spark.Response
 import spark.Spark
 import kotlin.math.max
 import kotlin.math.min
 
 object SettingsController {
-    fun saveBasic(request: Request, response: Response): Any {
+    fun saveSettings(request: Request, response: Response): Any {
         val params = request.paramsMap
+        val settings = duncteApis.getGuildSetting(request.guildId!!.toLong())
+
+        setBasic(params, settings)
+
+        // will return false ONLY if there are invalid rateLimits
+        if (!setModeration(params, settings)) {
```

Comment:
Should probably move this up so it can cancel all the sets when it returns (will probably not have a big impact)